backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/SubscriptionModelMapper.java

```diff
@@ -3,18 +3,34 @@
 import org.mapstruct.BeanMapping;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.Named;
 import org.mapstruct.NullValuePropertyMappingStrategy;
+import org.springframework.beans.factory.annotation.Autowired;
 
+import ca.gov.dtsstn.cdcp.api.service.AlertTypeService;
+import ca.gov.dtsstn.cdcp.api.service.domain.AlertType;
 import ca.gov.dtsstn.cdcp.api.service.domain.Subscription;
 import ca.gov.dtsstn.cdcp.api.web.v1.model.SubscriptionModel;
 import jakarta.annotation.Nullable;
 
 @Mapper(componentModel = "spring")
-public interface SubscriptionModelMapper {
+public abstract class SubscriptionModelMapper {
+
+	@Autowired
+	protected AlertTypeService alertTypeService;
 
 	@Nullable
 	@Mapping(target = "alertType", source = "alertType.code")
+	@Mapping(target = "add", ignore = true)
 	@BeanMapping(nullValuePropertyMappingStrategy = NullValuePropertyMappingStrategy.IGNORE)
-	SubscriptionModel toModel(@Nullable Subscription subscription);
+	public abstract SubscriptionModel toModel(@Nullable Subscription subscription);
+
+	@Nullable
+	@Mapping(target = "alertType", source = "alertType", qualifiedByName = { "toDomainAlerType" })
+	public abstract Subscription toDomain(@Nullable SubscriptionModel subscriptionModel);
 
+	@Named("toDomainAlerType")
+	protected AlertType toDomainAlerType(@Nullable String alertType) {
+		return alertTypeService.readByCode(alertType);
+	}	
```

Comment:
I don't love having this method in the mapper.. I feel like the mappers should be as dumb as possible.

Leave it as-is for now, but I might change this in a future PR.

I think the `SubscriptionService` should be responsible for calling the `AlertTypeService` and passing the `AlertType` to the mapper. This would reduce the coupling between classes, which I think is better overall.

Reply:
moved the calling AlertTypeService to SubscriptionService and mapped the alertTypeId